src/components/IOUConfirmationList.js

```diff
@@ -232,6 +274,44 @@ class IOUConfirmationList extends Component {
         return iouAmount !== sumAmount ? (amountPerPerson + difference) : amountPerPerson;
     }
 
+    /**
+    * Toggle selected option between selectedParticipant and unselectedParticipant.
+    * @param {Object} option
+    */
+    toggleOption(option) {
```

Comment:
I think we could clean up this function by changing the structure of this data. Rather than having two separate lists in state, (one for selected and one for unselected), what if we just save it as one list with each option having a bool for selected/unselected. That way this toggle function would just swap that bool and then call `setState` again. Easy peasy. 

This approach would change the logic in `getSections`, (where we are pushing the three lists) but I think that could be cleaned up as well, (and it wouldn't be too bad to use `map` or `reduce` to get each option into the list it needs to be in). 

Does that make sense? Do you like that idea? I'd be curious to see how many lines that saves us (I'm confident it'd make the file much shorter).

Reply:
Yes, will simplify the if else block in `getSections()`, saving us ~10 lines of code.

But then we'll have to get selected and unselected participants in `getSplits()`, `getSelectedOptions()`, `getSections()`, `toggleOption()`.

This adds 
- 8 lines of code
- Two new functions
- 8 loops over all the participants

And would make the code less _DRY_ .
So it makes sense to create two new states in my humble opinion.

Reply:
I ended up coding this up just to see, and it saved about 20 line of code. It makes `toggleOption()` look like this 
```js
toggleOption(option) {
        const participants = _.without(this.state.selectedParticipants, option);
        participants.push({
            ...option,
            selected: !option.selected,
        });
        this.setState({
            participants: participants
        });
}
```

I also added two helper functions
```js
    getSelectedParticipants() {
        return this.state.participants.filter(participant => participant.selected);
    }

    getUnselectedParticipants() {
        return this.state.participants.filter(participant => !participant.selected);
    }
```
It allowed me to take out `getFormattedUnselectedParticipants` completely. An interesting result was that it ends up iterating over the participants just one additional time bc the current `toggleOption` method iterates 4 times (`_.some` x 2, `_.without`, `_.reject`) and the new `toggleOption` method iterates just once. There are 4 places where we swap a reference to state for one of the helper methods I mentioned above, so overall the iteration counts are 4 vs 5. 

So at the end of the day, the trade-off is one more iteration for 20 lines of code. Personally, this is the way I'd write it, but it's totally OK if you'd like to keep the data structure you have now. Ultimately, there are lots of different ways to successfully write this, and I'm not going to force you one way or the other. Let me know if you'd like to see the diff and make the changes, if not, no worries.

Reply:
I agree with you on this one.
I've merged the two lists into one. Let me know what you think.